RMv4_Recipely/RMv4_Recipely/User Info/Profile Module/ProfileViewController.swift

```diff
@@ -0,0 +1,175 @@
+        setupTableView()
+    }
+
+    private func setupTableView() {
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.separatorStyle = .none
+
+        tableView.register(HeaderTableViewCell.self, forCellReuseIdentifier: Constants.headerTableViewCellIdentifier)
+        tableView.register(OptionsTableViewCell.self, forCellReuseIdentifier: Constants.optionsTableViewCellIdentifier)
+
+        view.addSubview(tableView)
+        tableView.dataSource = self
+        tableView.delegate = self
+
+        setTableViewConstraints()
+    }
+
+    private func setTableViewConstraints() {
+        tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor).isActive = true
+        tableView.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor).isActive = true
+        tableView.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor).isActive = true
+        tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor).isActive = true
+    }
+}
+
+/// Расширение вью методами протокола-интерфейса ProfileViewControllerProtocol
+extension ProfileViewController: ProfileViewControllerProtocol {
+    func reloadTableView() {
+        tableView.reloadData()
+    }
+
+    func showNameEditorAlert() {
+        let alert = UIAlertController(title: "Change your name and surname", message: nil, preferredStyle: .alert)
+
+        alert.addTextField()
+        alert.textFields?.first?.placeholder = "Name Surname"
+
+        let okAction = UIAlertAction(title: "Ok", style: .default) { [weak self] _ in
+            guard let text = alert.textFields?.first?.text else { return }
+            self?.presenter?.changeUserName(name: text)
+        }
+
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
+
+        alert.addAction(okAction)
+        alert.addAction(cancelAction)
+        alert.preferredAction = okAction
+
+        present(alert, animated: true)
+    }
+}
+
+/// Расширение вью методами UITableViewDataSource
+extension ProfileViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        tableViewSections.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch tableViewSections[section] {
+        case .headerCell:
+            return 1
+        case .cells:
+            return optionCells.count
+        }
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        switch tableViewSections[indexPath.section] {
+        case .headerCell:
+            guard let cell = tableView
+                .dequeueReusableCell(
+                    withIdentifier: Constants.headerTableViewCellIdentifier,
+                    for: indexPath
+                ) as? HeaderTableViewCell, let user = presenter?.user
```

Comment:
```suggestion
                ) as? HeaderTableViewCell, 
```
презентер присылает данные по запросу а не хранит их для доступа через свойство